stream-chat-android-ui-components/src/main/res/layout/stream_ui_activity_attachment_gallery.xml

```diff
@@ -6,86 +6,155 @@
-        tools:ignore="ContentDescription"
-        />
+        <TextView
+            android:id="@+id/attachmentDateTextView"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:gravity="center_horizontal"
+            android:textAppearance="@style/StreamUiTextAppearance.Footnote"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/attachmentActionsButton"
+            app:layout_constraintStart_toEndOf="@id/closeButton"
+            app:layout_constraintTop_toBottomOf="@id/userTextView"
+            app:layout_constraintVertical_chainStyle="packed"
+            tools:text="Sent yesterday at 4:08 PM"
+            />
+
+        <ImageView
+            android:id="@+id/attachmentActionsButton"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginEnd="@dimen/stream_ui_spacing_small"
+            android:background="?selectableItemBackgroundBorderless"
+            android:src="@drawable/stream_ui_ic_three_dots_menu"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            app:tint="@color/stream_ui_black"
+            tools:ignore="ContentDescription"
+            />
 
-    <androidx.constraintlayout.widget.Guideline
-        android:id="@+id/toolbarGuideline"
+        <View
+            android:layout_width="0dp"
+            android:layout_height="@dimen/stream_ui_divider_height"
+            android:background="@drawable/stream_ui_divider"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            />
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+    <androidx.viewpager2.widget.ViewPager2
+        android:id="@+id/galleryViewPager"
         android:layout_width="0dp"
         android:layout_height="0dp"
-        android:orientation="horizontal"
-        app:layout_constraintGuide_begin="@dimen/stream_ui_gallery_default_bar_height"
+        android:background="@color/stream_ui_white_snow"
+        app:layout_constraintBottom_toTopOf="@id/bottomBar"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/toolbar"
         />
 
-    <androidx.constraintlayout.widget.Group
-        android:id="@+id/toolBarGroup"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        app:constraint_referenced_ids="closeButton, title, subtitle, menuButton"
-        />
 
-    <io.getstream.chat.android.ui.gallery.AttachmentGallery
-        android:id="@+id/attachmentGallery"
+    <androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
This can be just a `Group`, but you can also use LinearLayout if you wanna use a ViewGroup.

Reply:
`LinearLayout` is slow with weights. Leaving this as is (in nested layout) for the sake of readability